### ISSUE TITLE
Grayscale option for thumbnails getting re-enabled automatically..

### ISSUE THREAD

**Issue by BlackSails7** [sniklaus/youtube-watchmarker#75]
I have the option disabled, but unfortunately it seems that every once per month or something, the option will get re-enabled again..

I was thinking that the add-on got updated and thus all preferences got reset to defaults? But I see that this add-on was last updated 8 months ago.. so this cant be it.

Btw I think that the grayscale option should be disabled by default. I'm confident that the majority of people don't prefer it, but not sure.

It has happened to me about 4 times so far. Last time it happened on December. I have noticed this happened with another addon also... Sponsor-block... multiple settings got changed... but it has only happened once with that. And I'm pretty sure it happened bcs of an update they made.

NOTE: I don't know whether the only option which gets re-enabled is grayscale.. since that's the only option I have disabled... and I cant reproduce the bug.. No idea why or when it happens.. but it does.

**Comment by BlackSails7** [comment]
It just happened again today, now that I restarted firefox... Didn't do anything else.

**Comment by sniklaus** [comment]
My apologies, but you are the only user reporting this issue and you reported that this issue happens with other extensions as well so I am under the impression that the issue does not originate from the Youtube Watchmarker. Do you have any other software on your computer that tampers with Firefox?

**Comment by BlackSails7** [comment]
I have been considering this as well, but nope.. this argument is not valid.

1. This has only happened with 1 other extension called SponsorBlock and it only happened once right after they added new options with an update. And never happened before or after this again.

2. I have installed exactly 69 addons, coincidentally another 69 userscripts in Violentmonkey and another 7 userscripts in Tampermonkey.

The java-userscripts do* also reset their settings whenever they are updated, but that's normal for userscripts... Users are not supposed to be changing the script. Whenever the script is automatically updated, all your modifications are gone.

But like I said, when it comes to addons, the only one that gives me this problem again and again is Youtube Watchmarker for some reason. I pay close attention to these things. So even if I was using an external app and this was the cause of this, then why is it that this is the only extension I have this issue with? Either way something is wrong with the saving method you're using. Otherwise all of my installed addons would have to lose their settings as well.

I just checked youtube thumbnails, and grayscale was disabled... then closed firefox... then opened it again.. went on youtube and the thumbnails were gray again meaning it got reset again... Then I disabled grayscale in settings once again... closed firefox... opened it.. went on youtube and the grayscale stayed disabled as it should.. I tried this one more time with the same result..

So to me.. it seems that it happens whenever I restart firefox after 1-2 weeks of not closing it. Btw I'm using Firefox 91.0.1, I have not updated for months. Also.. I only restart Firefox once or twice per month.. I always hibernate my PC, so I don't have to close everything.

I don't use any external software on firefox at all. I just looked through all my installed apps.

The only external software that I rarely used on Firefox was CCleaner, but since I don't close firefox often.. I have disabled Firefox from the options, so it doesn't clean this...

And just in case... to prove the point that it has nothing to do with it.. I just closed firefox, opened CCleaner and cleared firefox (cache etc) like I have done before in the past.. and the grayscale option didn't get reset.. It stayed disabled as it should.

I don't get why this is happening. Perhaps most other users have grayscale enabled since this is the default? And those few who had it disabled, didn't pay attention when it gets enabled automatically so they also got used to it being this way and didn't bother report this? I mean.. you only have 14 issues reported... perhaps not many people are into reporting this in this case.

It took me over 1 year before I reported this issue.

**Comment by sniklaus** [comment]
The setting you are referring to is called `extensions.Youwatch.Visualization.boolGrayout` internally.

It gets set to `true` if it is `null` which happens when the extension is used for the first time.
https://github.com/sniklaus/youtube-watchmarker/blob/b959f02f14707191e1220172789de486cf35bf8b/background.js#L1463-L1465

If it is `true`, it will add CSS to make the thumbnails gray on any existing/new Youtube tab.
https://github.com/sniklaus/youtube-watchmarker/blob/b959f02f14707191e1220172789de486cf35bf8b/background.js#L1562-L1566

If you click the checkbox in the settings the value will be set to `true` if it is `false` and vice versa.
https://github.com/sniklaus/youtube-watchmarker/blob/427e6b201d5de6fae70f9e9d3644deb9262246cc/content/index.js#L458

When you open the settings then it is used to determine whether the checkbox in the UI should be set or not.
https://github.com/sniklaus/youtube-watchmarker/blob/427e6b201d5de6fae70f9e9d3644deb9262246cc/content/index.js#L475-L486

That's it, those are all the occurrences of `boolGrayout`: https://github.com/sniklaus/youtube-watchmarker/search?q=boolGrayout

So in summary, the only time `extensions.Youwatch.Visualization.boolGrayout` gets changed is when the extension is opened for the first time or if you click the corresponding checkbox in the settings.

**Comment by BlackSails7** [comment]
So then I think the problem might be that some addon that I'm using is not compatible with Youtube Watchmarker.. I've checked the scripts, and I don't see anything that could mess with this. So it's probably some addon.

I have also disabled the **Hide the progressbar...underneath watched videos.** option just to see whether this will be reset whenever grayscale does. Because I suspect that all settings get reset, not just grayscale.

I have also messed with a lot of settings on firefox in about:config, dunno w

I'll experiment more and try to figure out exactly what's causing this. If I figure it out, I'll let you know.

**Comment by sniklaus** [comment]
Sounds good, I am curious to hear about the root cause should you ever find one. I am closing this issue for now since it doesn't seem like the Youtube Watchmarker is to blame here. Please let me know if you find anything that suggest otherwise though so that I can reopen this case.
